### ISSUE TITLE
Style/TernaryParentheses require_parentheses_when_complex false positive

### ISSUE THREAD

**Issue by dvandersluis** [rubocop-hq/rubocop#6374]
The `Style/TernaryParentheses` cop when `EnforcedStyle: require_parentheses_when_complex` is set considers simple comparisons (where there is just a single term) to be "complex", and adds parens on auto complete.

```ruby
foo == bar ? 1 : 2
```

--------

## Expected behavior

No offense is registered, autocorrect doesn't add parens.

## Actual behavior

Adds an offense, and autocorrects to `(foo == bar) ? 1 : 2`

## Steps to reproduce the problem

See example above.

## RuboCop version

Include the output of `rubocop -V` or `bundle exec rubocop -V` if using Bundler. Here's an example:

```
$ [bundle exec] rubocop -V
0.59.2 (using Parser 2.5.1.2, running on ruby 2.4.4 x86_64-darwin17)
```

**Comment by koic** [comment]
It seems necessary to think about "complex" handled by this cop.

I looked it, this cop seems to handle using comparison operators as "complex".

```ruby
  context 'configured for parentheses on complex and there are no parens' do
    let(:cop_config) do
      { 'EnforcedStyle' => 'require_parentheses_when_complex' }
    end

    (snip)

    context 'with complex condition' do

      (snip)

      it_behaves_like 'code with offense',
                      'foo = bar < baz ? a : b',
                      'foo = (bar < baz) ? a : b'
```

https://github.com/rubocop-hq/rubocop/blob/v0.59.2/spec/rubocop/cop/style/ternary_parentheses_spec.rb#L343-L345

If it conform to this spec, I think that it probably is not enough example code in documentation.

>  simple comparisons (where there is just a single term)

On the other hand, I think the above view makes sense.

I'd like to ask other opinions :-)

**Comment by Drenmi** [comment]
I am not a fan of this configuration setting. Firstly, “complex” is arbitrary, secondly, having if-then style rules generally works poorly, because people need to memorize the different cases, increasing cognitive overhead instead of reducing it.

**Comment by dvandersluis** [comment]
I do think it's helpful to make code a little more readable when there are long ternaries (which should really be refactored out, but that's a different story). But it is definitely a little arbitrary right now what is considered complex.

Looking at the tests for this cop, a lot of them involve assignment to a ternary, but I'm also interested in a ternary without assignment like my original example. I don't know if there _should_ be a difference between ternaries used in assignment and not.
